### ISSUE TITLE
Handling of documentation

### ISSUE THREAD

**Issue by s-poll** [HPSCTerrSys/eCLM_static_file_workflow#2]
Hi all, 

I just saw that the new documentation in the eCLM repository is ready to be released [eCLM PR #38](https://github.com/HPSCTerrSys/eCLM/pull/38). Having a documentation (nicely looking) was long overdue. However, I would also favor to have in-repo documentation, as this is the easiest way to ensure synchronization between documentation and source code. This is now the case for the eCLM part, but also the part of the `case creation`, which belongs to the `eCLM_static_file_workflow` is part of it. 

I would be in favor to move the section `case creation` to this repository and just reference to the workflow-documentation in the elm-documentation. 

What are your thoughts on that? Explicitly mentioning @odombro, @kvrigor and @AGonzalezNicolas as you are - to my knowledge - involved in the creation of the documentation.

**Comment by AGonzalezNicolas** [comment]
I agree that the `case creation` section should go into the repository `eclm_static_file_workflow`.

**Comment by kvrigor** [comment]
I'm also in favor of built-in docs as it makes a code repository self-contained. However from the user's POV, I think it's ideal to access all eCLM-related information from a single place, which is the eCLM doc website itself. A workaround is to link the case creation docs to this repo, but this will have the following consequences:

- The doc CI build workflow will become more complicated since the content would have to be fetched from multiple repos.
- Editing docs will be more complicated as pull requests would have to be raised on separate repos.
- A separate doc website could be set up for this repo, but this would fragment the eCLM docs which IMO is an unnecessary complexity.

I believe we should avoid additional maintenance burden as much as possible. Also, we could encourage more user contributions if the editing workflow is much easier.

How should we resolve the dilemma of having self-contained docs vs. a centralized one? One solution could be to `git submodule` this repo under eCLM. This minimizes maintenance overhead but doesn't address the complicated editing workflow, which may be an acceptable compromise. Any other ideas?

**Comment by odombro** [comment]
Would it be an option to merge these two repos into one? 
I have mostly left out the ICON-specific part of the case creation workflow for the eCLM documentation but it could be added again to not miss this part for ICON users.

**Comment by kvrigor** [comment]
> Would it be an option to merge these two repos into one?

I'm OK with this. Since domain and surface files must be compatible with eCLM inputs, I think it's only natural to embed such tools that generate these files within eCLM itself. Also this allows to improve the existing case creation scripts in a way that takes advantage of eCLM model codes (e.g. directly calling Fortran subroutines from Python).

**Comment by DCaviedesV** [comment]
I have a different opinion here.
In my view the model's code is one thing. Case creating, workflow management, and anything else, are different projects and thus different repositories. 
They are certainly related and intertwined. 
However, keeping everything together, although it provides a one-stop-shop, also blurs the differences between these aspects for the user (where does the model start/end, where do the workflows start/end?), and for the maintainers (who is responsible for which part, if it is all in the same bag?). 
In the end you don't really need the `eCLM_static_file_workflow` to operate `eCLM`... it is convenient, but it is an additional tool, not a core part of the solver. The analogy for this is the TSMP code and the TSMP workflow manager, clearly not the same thing, but all works together.

**Comment by AGonzalezNicolas** [comment]
I am with Johannes and Daniel: separate case creation and source code.

**Comment by kvrigor** [comment]
I'd also vote to close the https://github.com/HPSCTerrSys/eCLM/pull/38 and do the next round of improvements on a new pull request.

**Comment by kvrigor** [comment]
With @mvhulten's recent work to the static file workflow, I think it's a good time to revisit this issue. Should the [eCLM case creation guide] be replaced by a link to this [repo's README](https://github.com/HPSCTerrSys/eCLM_static-file-generator/blob/main/README.md)? I have a feeling that the [eCLM case creation guide] is already outdated due to the developments here, but I'm not sure.

[eCLM case creation guide]: https://hpscterrsys.github.io/eCLM/users_guide/case_creation/README.html

**Comment by mvhulten** [comment]
The README here is not finished yet, but especially given all the associated changes in the scripts, it is certainly more up to date than the [eCLM case creation guide](https://hpscterrsys.github.io/eCLM/users_guide/case_creation/README.html).  Furthermore, there are a few bits (like the figure—does anyone have the vector source of [this graph](https://github.com/HPSCTerrSys/eCLM/blob/master/docs/users_guide/images/Build_custom_input.png)?) that are amendable to integrate into our new guide.

But Stefan and I were thinking about creating [a docs directory](https://github.com/HPSCTerrSys/eCLM_static-file-generator/tree/docs), just like [this one](https://github.com/HPSCTerrSys/eCLM/tree/master/docs) in the eCLM repo.  It would be nice if this could be linked into (with the effect of looking as a proper section of) the eCLM documentation.

**Comment by kvrigor** [comment]
> does anyone have the vector source of [this graph](https://github.com/HPSCTerrSys/eCLM/blob/master/docs/users_guide/images/Build_custom_input.png)?) that are amendable to integrate into our new guide.

I suggest redrawing the graph using the online drawing tool [Excalidraw](https://excalidraw.com). This tool is quite easy to use.

> But Stefan and I were thinking about creating [a docs directory](https://github.com/HPSCTerrSys/eCLM_static-file-generator/tree/docs), just like [this one](https://github.com/HPSCTerrSys/eCLM/tree/master/docs) in the eCLM repo. It would be nice if this could be linked into (with the effect of looking as a proper section of) the eCLM documentation.

+1. I can modify the eCLM online doc generator such that it would automatically pull assets under `eCLM_static-file-generator/docs` and then renders them as a part of the [eCLM doc website](https://hpscterrsys.github.io/eCLM/INDEX.html).

**Comment by mvhulten** [comment]
I changed my opinion about the `doc/` directory and putting it on <https://hpscterrsys.github.io/>.

Even though it would still live in the same repo, there will be many files (each section) and they will be small because the README is and should be and should stay relatively short and straighforward.  I think the static file generators will be different in length and structure compared to e.g. a workflow engine.  Editting (and reading) a single Markdown/HTML page is easier.  I'd opt to leave it as it is, certainly as long as it is not finished.  It would be a distraction to change this.

**Comment by jjokella** [comment]
With #67 solved and #63 merged, I close this issue for the moment.
